File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        
        for (int num: arr) {
            int count = 0;
            while (true) {
                if (num >= 50 && num % 2 == 0) {
                    num /= 2;
                    answer = Math.max(answer, ++count);
                }
                else if (num < 50 && num % 2 != 0) {
                    num = 2*num + 1;
                    answer = Math.max(answer, ++count);
                }
                else
                    break;
            }
        }
     
        return answer;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = 0
    
    for num in arr:
        count = 0
        while(1):
            if num >= 50 and num % 2 == 0:
                num /= 2
                count += 1
                answer = max(answer, count)
            elif num < 50 and num % 2 != 0:
                num = 2*num + 1
                count += 1
                answer = max(answer, count)
            else:
                break
                
    return answer